backend/src/packages/drivers/driver.repository.ts

```diff
@@ -36,22 +36,62 @@ class DriverRepository implements IRepository {
 
     const finalQuery = queries.length === 1 ? queries[0] : and(...queries);
 
-    return this.db
+    const drivers = await this.db
       .driver()
-      .query.drivers.findMany({ where: finalQuery })
+      .query.drivers.findMany({ where: finalQuery, with: { user: true } })
       .execute();
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    return drivers.map(({ createdAt, updatedAt, ...pureDriver }) =>
+      DriverEntity.initialize({
+        ...pureDriver,
+        createdAt: createdAt.toISOString(),
+      }).toObject(),
+    );
   }
 
   public async findAllByBusinessId(
     businessId: number,
   ): Promise<DriverEntity[]> {
     const drivers = await this.db
       .driver()
-      .select()
-      .from(this.driverSchema)
-      .where(eq(this.driverSchema.businessId, businessId));
+      .query.drivers.findMany({
+        where: eq(this.driverSchema.businessId, businessId),
+        with: { user: true },
+      })
+      .execute();
 
-    return drivers.map((it) => DriverEntity.initialize(it));
+    return drivers.map(({ createdAt, ...pureDriver }) =>
+      DriverEntity.initialize({
+        ...pureDriver,
+        createdAt: createdAt.toISOString(),
+      }),
+    );
+  }
+
+  public async findPageOfDrivers(
+    businessId: number,
+    pageIndex: number,
+    pageSize: number,
+  ): Promise<DriverEntity[]> {
+    const offset = pageIndex * pageSize;
+    const drivers = await this.db
```

Comment:
won't it work?

 return await this.db
      .driver()
      .select()
      .from(this.driverSchema)
      .where(eq(this.driverSchema.businessId, businessId))
      .orderBy(desc(this.driverSchema.createdAt))
      .offset(index)
      .limit(query.pageSize);

Reply:
We need to get driver info with user info, but in this case, we get just driver info. And second point in this service we have a strange structure when we should initialize in repo object and then convert to object  in service